tools/lsp/preview.rs

```diff
@@ -124,6 +125,37 @@ pub fn poll_once<F: std::future::Future>(future: F) -> Option<F::Output> {
     }
 }
 
+fn invalidate_contents(url: &lsp_types::Url) {
+    i_slint_core::debug_log!("PREVIEW: Invalidate {url}");
+    let component = {
+        let cache = CONTENT_CACHE.get_or_init(Default::default).lock().unwrap();
+        (cache.dependency.contains(url) && cache.ui_is_visible)
+            .then_some(())
+            .and_then(|_| cache.current_component())
+    };
+
+    if let Some(document_cache) = document_cache() {
+        // Do not add random things to the docuemnt cache!
+        if document_cache.get_document(url).is_some() {
```

Comment:
The preview also has a concept of what files are actually direct dependency of the currently previewed document. We should use that. (because IIRC, the document cache can have all files, even the ones not used by the preview)